Update pac4j to 6.1.1

sake92 opened the pull request:
Not an expert here, and I will probably need some guidance (there a few TODOs I left).  
I've mostly looked into https://github.com/pac4j/javalin-pac4j and JEE implementations since they seem similar to undertow.

I will try to add some tests too, adapt from javalin-pac4j

leleuj commented:
Please fix the `ci.yml` file to release 17 for the build to start.

Overall, LGTM. A few comments to address. Can you update at best the undertow-pac4j-demo as well? Thanks

sake92 commented:
Updated the CI now.  
Which comments I need to address?
I will update the demo after we get a new release, no problem.

leleuj reviewed (changes requested):
Please the requested changes

sake92 commented on src/main/java/org/pac4j/framework/adapter/FrameworkAdapterImpl.java:
```diff
@@ -0,0 +1,24 @@
+import org.pac4j.undertow.http.UndertowHttpActionAdapter;
+
+// found on classpath by pac4j
+public class FrameworkAdapterImpl extends DefaultFrameworkAdapter {
```
Added.

sake92 commented on src/main/java/org/pac4j/undertow/handler/SecurityHandler.java:
```diff
@@ -73,18 +65,24 @@ public static HttpHandler build(final HttpHandler toWrap, Config config, final S
+                this.matchers,
+                new UndertowParameters(exchange)
+        );
+        // TODO else reject with 401 or?
```
Ok, I didnt really understand how it works.  
So that callback is triggered on successful auth, if I understood correctly..

sake92 commented:
@leleuj I've addressed your comments, please have a look again.  
I've tested this locally with a FormClient, works fine.

I'll try to add some real tests in future PRs.

leleuj commented:
I just merged your PR. Thanks for the contribution.
